### PR TITLE
Use SVGs over PNGs for the sidebar when possible

### PR DESCRIPTION
**Pull request by janfaracik**
### Proposed changelog entries

* Use SVGs over PNGs for the sidebar when possible
* Breadcrumb bar/logo/menu items are now correctly aligned on the left together
* Move old `war/images` folder to `webapp` so they can be used in frontend - the SVGs are now in the `webapp/images/svgs` folder

### Screenshots

**Old**
image

**New**
image

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
the new view icon is different in the two screenshots.

When I start this with core-pr-tester all the default icons are looking better except the new view icon which is still using the same as the old one.

---

Could you explain the "Move old images folder to accessible scalable folder" change a bit more please? There's some `.pngs` added in there, why not use the `svgs` folder?

**Review by oleg-nenashev — COMMENTED** [review]
I am not too concerned about the View icon. Currently we have a view vs. folder confusion, and ideally we should change the icon all together. Maybe a generic link like https://materialdesignicons.com/icon/view-agenda would be better. Anyway, it rather should be done as a separate PR

**Comment by janfaracik** [comment]
> the new view icon is different in the two screenshots.
> 
> When I start this with core-pr-tester all the default icons are looking better except the new view icon which is still using the same as the old one.
> 
> Could you explain the "Move old images folder to accessible scalable folder" change a bit more please? There's some `.pngs` added in there, why not use the `svgs` folder?

Ah that's my bad, the screenshot in the old version is from a different build of Jenkins using a different icon for some reason (maybe a plugin? I'm not too sure). This is the icon currently on master:

image

I've just rebuilt Jenkins and it's using the new icon on my side, not too sure what's going on there.

--

I'll update the PR post 👍 There were a slew of SVGs tucked away in `war/images` which I imagine weren't accessible to the frontend, so I've moved them to a new folder in `webapp/images/scalable` - I'm more than happy to move them to the existing SVGs folder.

**Comment by timja** [comment]
If you run the PR with the core-pr-tester which guarantees a clean slate (skip plugin installation), then you get the same icon as original

```command
docker run --rm -ti -p 8080:8080 -e ID=5367 -e MERGE_WITH=master jenkins/core-pr-tester
```

File: core/src/main/java/org/jenkins/ui/icon/IconSet.java [hunk 1]
```diff
@@ -469,6 +469,16 @@ private static String toNormalizedIconUrl(String url) {
         icons.addIcon(new Icon("icon-up icon-md", "24x24/up.png", Icon.ICON_MEDIUM_STYLE));
```
Comment (timja) [inline core/src/main/java/org/jenkins/ui/icon/IconSet.java:469]: can't we remove these old icon's (at least the medium style)?

**Comment by timja** [comment]
Hmm actually with core-pr-tester it's fine, let me check with plugins

**Review by timja — APPROVED** [review]
> Ah that's my bad, the screenshot in the old version is from a different build of Jenkins using a different icon for some reason (maybe a plugin? I'm not too sure). This is the icon currently on master

You were right, if you install Jenkins via setup wizard with the suggested plugins then you get the png image. Without installing plugins you get the correct one. Another plugin is overriding it most likely the cloudbees-folders plugin.

We can sort that in a separate PR

File: core/src/main/java/org/jenkins/ui/icon/IconSet.java (hunk 1 above)
Comment (janfaracik) [reply core/src/main/java/org/jenkins/ui/icon/IconSet.java:469]: I had left them there as I wasn't sure if they were needed for backwards compat - more than happy to remove them though, am I good to remove any .gifs of the same names too?

Comment (timja) [reply core/src/main/java/org/jenkins/ui/icon/IconSet.java]: Yes

**Comment by timja** [comment]
@janfaracik I pushed a couple of commits, I hope you don't mind,

1. fix unit test failure where the old icons were being checked
2. icon-help was changed back to the old icon un-intentionally, as the new svg was overriding the proper one
3. re-indented the changes you made to be consistent with the rest of the file in IconSet

File: core/src/main/java/org/jenkins/ui/icon/IconSet.java (hunk 1 above)
Comment (janfaracik) [reply core/src/main/java/org/jenkins/ui/icon/IconSet.java]: Awesome, those are all gone now 🚀 

Are the actual image files good to be deleted as well or would they need to stay for backwards compat?

**Review by uhafner — APPROVED** [review]
Looks good, thanks!

Maybe it would be helpful as a followup to describe somewhere what plugin authors should do to replace their PNG icons in the sidebar actions with SVG icons.

**Comment by timja** [comment]
@janfaracik can you take a look at the tests please: https://github.com/jenkinsci/jenkins/pull/5663/checks?check_run_id=3344683253

Looks like a regression in one of your last commits

**Comment by janfaracik** [comment]
> @janfaracik can you take a look at the tests please: https://github.com/jenkinsci/jenkins/pull/5663/checks?check_run_id=3344683253
> 
> Looks like a regression in one of your last commits

All should be passing now :)

**Review by oleg-nenashev — APPROVED** [review]
No objections from me. Not sure whether it will be all good w.r.t compatibility, but I am definitely +1 for simplifying

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Comment by timja** [comment]
Caused 
image

On plugin manager pages, e.g.
https://$JENKINS_SERVER/pluginManager/installed

@janfaracik any chance you can take a look?

**Comment by janfaracik** [comment]
> Caused
> image
> 
> On plugin manager pages, e.g.
> https://$JENKINS_SERVER/pluginManager/installed
> 
> @janfaracik any chance you can take a look?

Fixed here - https://github.com/jenkinsci/jenkins/pull/5690

Thanks for the heads up :)

**Comment by timja** [comment]
Probably caused https://issues.jenkins.io/browse/JENKINS-66522 looks to be a conflict in versions with some plugin, but maven-plugin appears to have it setup to pull from core correctly

**Comment by timja** [comment]
Caused https://issues.jenkins.io/browse/JENKINS-66565

**Pull request merged**